### PR TITLE
🤓 Implemented feedback from old PRs

### PR DESCRIPTION
**Pull request by dees040**

### CODE REVIEW

File: src/Http/Controllers/Controller.php [hunk 1]
```diff
@@ -58,4 +67,32 @@ protected function getUserFromRoute(): Model
...
+    /**
+     * Tries to retrieve the model.
+     *
+     * @param  mixed  $value
+     * @param  string|null  $modelClass
+     * @return \Illuminate\Database\Eloquent\Model
+     *
+     * @throws \Illuminate\Database\Eloquent\ModelNotFoundException
+     */
+    public function getModel($value, string $modelClass = null): Model
```
Comment (dees040) [inline src/Http/Controllers/Controller.php:80]: Code got improved in [this PR](https://github.com/owowagency/laravel-gossip/pull/13/files#diff-06fe6c5c07e5f6748059572da347bdf0fd666746b5d98818eed6fd7bebf3f8cbR12).

File: src/Managers/MessageManager.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+class MessageManager
+{
+    /**
+     * Create a new Eloquent Collection instance of messages.
+     *
+     * @param  array  $models
+     * @return \OwowAgency\Gossip\Support\Collection\MessageCollection
+     */
+    public static function createCollection(array $models): MessageCollection
```
Comment (thomasowow) [inline src/Managers/MessageManager.php:16]: Still think this belongs in controller. This is just like going anal, going vag and than checking for condom.

File: src/Models/Message.php [hunk 3]
```diff
@@ -76,12 +76,9 @@ public function conversation(): BelongsTo
      * @param  \OwowAgency\Gossip\Models\Conversation  $conversation
      * @return \Illuminate\Database\Eloquent\Builder
      */
-    public function scopeOfConversation($query, Conversation $conversation)
+    public function scopeOfConversation($query, $conversation)
```
Comment (thomasowow) [inline src/Models/Message.php:79]: Type hint it with Eloquent model?

File: src/Http/Controllers/Conversations/ConversationController.php [hunk 4]
```diff
@@ -17,7 +25,7 @@ public function index(): JsonResponse
     {
         $this->authorize('viewAny', [Conversation::class]);
 
-        $conversations = Conversation::withRelations(3)
+        $conversations = Conversation::withDefaultRelations(3)
```
[the remark this answers is not in the document]
Comment (dees040) [reply src/Http/Controllers/Conversations/ConversationController.php:28]: You can easily reuse this via the trait

File: src/Managers/MessageManager.php [hunk 5]
```diff
@@ -4,33 +4,26 @@
...
-        );
-
-        if ($markAsRead) {
-            $collection->markAsRead(Auth::user());
+    public static function handleMessageMarking(
+        MessageCollection $collection,
+        $markAsRead,
+        HasConversationContract $user = null
+    ): MessageCollection {
+        if (filter_var($markAsRead, FILTER_VALIDATE_BOOLEAN)) {
```
Comment (thomasowow) [inline src/Managers/MessageManager.php:25]: Not sure but I think if you typehint the parameter it might be converted to `boolean`.

**Review by thomasowow — APPROVED** [review]

File: src/Managers/MessageManager.php (hunk 5 above)
Comment (dees040) [reply src/Managers/MessageManager.php:25]: You're right, nice tip! 🚀

**Pull request merged**